src/dom.ts

```diff
@@ -0,0 +1,95 @@
+// Types taken from https://www.w3.org/TR/webtransport
+// These can be removed when they are added to the default typescript types
+
+export interface WebTransportDatagramStats {
+  timestamp: number
+  expiredOutgoing: bigint
+  droppedIncoming: bigint
+  lostOutgoing: bigint
+}
+
+export interface WebTransportStats {
+  timestamp: number
+  bytesSent: bigint
+  packetsSent: bigint
+  packetsLost: bigint
+  numOutgoingStreamsCreated: number
+  numIncomingStreamsCreated: number
+  bytesReceived: bigint
+  packetsReceived: bigint
+  smoothedRtt: number
+  rttVariation: number
+  minRtt: number
+  datagrams: WebTransportDatagramStats
+}
+
+export interface WebTransportCloseInfo {
+  closeCode: number
+  reason: string
+}
+
+export interface WebTransportDatagramDuplexStream {
+  readable: ReadableStream<Uint8Array>
+  writable: WritableStream<Uint8Array>
+  readonly maxDatagramSize: number
+  incomingMaxAge?: number
+  outgoingMaxAge?: number
+  incomingHighWaterMark: number
+  outgoingHighWaterMark: number
+}
+
+export interface WebTransportBidirectionalStream {
+  readonly readable: ReadableStream<Uint8Array>
+  readonly writable: WritableStream<Uint8Array>
+}
+
+export interface  WebTransportSendStreamStats {
+  timestamp: number
+  bytesWritten: bigint
+  bytesSent: bigint
+  bytesAcknowledged: bigint
+}
+
+export interface WebTransportSendStream extends WritableStream<Uint8Array> {
+  getStats: () => Promise<WebTransportSendStreamStats>
+}
+
+export interface WebTransportReceiveStreamStats {
+  timestamp: number
+  bytesReceived: bigint
+  bytesRead: bigint
+}
+
+export interface WebTransportReceiveStream extends ReadableStream<Uint8Array> {
+  getStats: () => Promise<WebTransportReceiveStreamStats>
+}
+
+export interface WebTransportHash {
+  algorithm: string
+  value: BufferSource
+}
+
+export interface WebTransportOptions {
+  allowPooling?: boolean
+  requireUnreliable?: boolean
+  serverCertificateHashes?: WebTransportHash[]
+}
+
+export interface WebTransport {
+  getStats: () => Promise<WebTransportStats>
+  readonly ready: Promise<void>
+  readonly reliability: WebTransportReliabilityMode
```

Comment:
This is legacy, comment, it makes also no sense in the spec

Reply:
There's some more context here: https://github.com/w3c/webtransport/issues/107 but the 'reliability' property tells us if we've fallen back to http2/tcp for the connection so this should probably stay in?

Reply:
Well, they dropped http/2 as well as quic support entirely in favor of http/3. So it was just not deleted.

Reply:
Got a link?  I'm still learning about this stuff..

Reply:
A single link is hard to find...
Here is an old spec 
https://datatracker.ietf.org/doc/html/draft-vvv-webtransport-quic-00
called quictransport and there were many iterations of the draft. Furthermore you see it in the quiche source code, there are still relicts. But you also see that is not capable of http2 webtransport at all. So the best reference is the quiche source code and in the web transport spec the links to the specification of the underlying protocols at the end. See § 10 and its references.